File: lab-2.3/c++/TextLine.cpp
#include "TextLine.h"

TextLine::TextLine(const std::string& line) : content(line) {}

const std::string& TextLine::getContent() const {
    return content;
}

std::string TextLine::getDigitsString() const {
    std::string digits;
    for (char c : content) {
        if (isdigit(c)) {
            digits += c;
        }
    }
    return digits;
}

File: lab-2.3/c#/TextLine.cs
using System;
using System.Linq;

namespace TextLibrary
{
    public class TextLine : IDigits
    {
        private readonly string content;

        public TextLine(string line)
        {
            content = line;
        }

        public string Content => content;

        public string GetDigitsString()
        {
            return new string(content.Where(char.IsDigit).ToArray());
        }
    }
}